tests/Common/BaseTestCase.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace DDTrace\Tests\Common;
+
+use DDTrace\Tests\Common\MultiPHPUnitVersionAdapter;
+use DDTrace\Log\Logger;
+use DDTrace\Tests\DebugLogger;
+use DDTrace\Util\Versions;
+
+abstract class BaseTestCase extends MultiPHPUnitVersionAdapter
+{
+    public static function ddSetUpBeforeClass()
+    {
+    }
+
+    public static function ddTearDownAfterClass()
+    {
+    }
+
+    protected function ddSetUp()
+    {
+    }
+
+    protected function ddTearDown()
+    {
+        \Mockery::close();
+        Logger::reset();
+        \dd_trace_internal_fn('ddtrace_reload_config');
+    }
+
+    protected function matchesPhpVersion($version)
+    {
+        return Versions::phpVersionMatches($version);
+    }
+
+    /**
+     * Sets and return a debug logger which accumulates log messages.
+     * @return DebugLogger
+     */
+    protected function withDebugLogger()
+    {
+        $logger = new DebugLogger();
+        Logger::set($logger);
+        return $logger;
+    }
+
+    public function composerUpdateScenario($workingDir)
```

Comment:
It looks like this was moved from the old `BaseTestCase`, but this seems to be called exclusively from `ComposerInteroperabilityTest`, so maybe we could move it there? It also might be good to send `$workingDir` through `escapeshellarg()` just in case. :)